### PR TITLE
[Core]ReadmaterialsUtility check for value-overwrites

### PR DESCRIPTION
**Pull request by philbucher**
plus removing the destructor (not needed)

### CODE REVIEW

File: kratos/utilities/read_materials_utility.cpp [hunk 1]
```diff
@@ -183,12 +184,15 @@ void ReadMaterialsUtility::AssignPropertyBlock(Parameters Data)
         if (KratosComponents<Variable<double> >::Has(variable_name)) {
             const Variable<double>& variable = KratosComponents<Variable<double>>().Get(variable_name);
             p_prop->SetValue(variable, value.GetDouble());
+            CheckIfOverwrittingValue(*p_prop, variable, value.GetDouble());
```
Comment (armingeiser) [inline kratos/utilities/read_materials_utility.cpp]: Shouldn't this happen before you set the value?

**Review by loumalouomega — DISMISSED** [review]
Not a class anymore?

File: kratos/utilities/read_materials_utility.cpp (hunk 1 above)
Comment (philbucher) [reply kratos/utilities/read_materials_utility.cpp]: :D ups

**Comment by philbucher** [comment]
> Not a class anymore?

it still is, I just thought this function is only used in this cpp so why mess up the header

**Comment by loumalouomega** [comment]
OK

**Review by loumalouomega — APPROVED** [review]

**Review by loumalouomega — APPROVED** [review]

**Comment by pooyan-dadvand** [comment]
> it still is, I just thought this function is only used in this cpp so why mess up the header

To be honest not very good practice. I would at least putting such functions in Internals namespace

File: kratos/utilities/read_materials_utility.cpp [hunk 2]
```diff
@@ -18,11 +18,19 @@
 // Project includes
 #include "utilities/read_materials_utility.h"
 
-namespace Kratos
-{
+namespace Kratos {
+namespace {
```
Comment (philbucher) [inline kratos/utilities/read_materials_utility.cpp:22]: @pooyan-dadvand I did do that

Comment (pooyan-dadvand) [reply kratos/utilities/read_materials_utility.cpp:22]: Sorry, I have not noticed that :+1:

**Pull request merged**